src/main/java/org/jboss/remoting3/EndpointImpl.java

```diff
@@ -241,6 +258,8 @@ static EndpointImpl construct(final EndpointBuilder endpointBuilder) throws IOEx
             endpoint = new EndpointImpl(xnioWorker, false, endpointName, connectionOptions.isEmpty() ? Collections.emptyMap() : connectionOptions);
         }
         boolean ok = false;
+        //Set default option map here so we do not modify the constructor
```

Comment:
@dmlloyd  can you please double check this, is changing the constructor really a problem here? The code would be more straightforward if we could add defaultOptionMap field as final and set it in the constructor...

Reply:
Yeah setting it in the constructor is no problem.  The constructor is not public.

Reply:
okay, so @tmiyargi, please move that field as a final and set it in EndpointImpl as an argument, that will be cleaner, btw, thanks a lot for your fix!